test/functional/pdo_sqlsrv/PDO61_BindColumn1.phpt

```diff
@@ -5,124 +5,124 @@ Verification for "PDOStatement::bindColumn".
-	LogInfo(1, "Testing fetchColumn() ...");
-	$stmt1 = ExecuteQuery($conn1, "SELECT COUNT(idx) FROM [$tableName]");
-	var_dump($stmt1->fetchColumn());
-	unset($stmt1);
-
-	LogInfo(2, "Testing fetchAll() ...");
-	$stmt1 = PrepareQuery($conn1, "SELECT idx, txt FROM [$tableName] ORDER BY idx");
-	$stmt1->execute();
-	$data = $stmt1->fetchAll(PDO::FETCH_COLUMN|PDO::FETCH_UNIQUE);
-	var_dump($data);
-
-	LogInfo(3, "Testing bindColumn() ...");
-	$stmt1->bindColumn('idx', $idx);
-	$stmt1->bindColumn('txt', $txt);
-	$stmt1->execute();
-	while ($stmt1->fetch(PDO::FETCH_BOUND))
-	{
-		var_dump(array($idx=>$txt));
-	}
-
-	LogInfo(4, "Testing bindColumn() with data check ...");
-	$id = null;
-	$val = null;
-	$data = array();
-	$index = 0;
-	if (!$stmt1->bindColumn(1, $id, PDO::PARAM_INT))
-	{
-		LogError(5, "Cannot bind integer column", $stmt1);
-	}
-	if (!$stmt1->bindColumn(2, $val, PDO::PARAM_STR))
-	{
-		LogError(5, "Cannot bind string column", $stmt1);
-	}
-	$stmt1->execute();
-	while ($stmt1->fetch(PDO::FETCH_BOUND))
-	{
-		$data[] = array('id' => $id, 'val' => $val);
-		printf("id = %s (%s) / val = %s (%s)\n",
-			var_export($id, true), gettype($id),
-			var_export($val, true), gettype($val));
-	}
-	unset($stmt1);
-	$stmt1 = ExecuteQuery($conn1, "SELECT idx, txt FROM [$tableName] ORDER BY idx");
-	while ($row = $stmt1->fetch(PDO::FETCH_ASSOC))
-	{
-		if ($row['idx'] != $data[$index]['id'])
-		{
-			LogInfo(6, "Data corruption for integer column in row $index");
-		}
-		if ($row['txt'] != $data[$index]['val'])
-		{
-			LogInfo(6, "Data corruption for string column in row $index");
-		}
-		$index++;
-	}
-
-
-	// Cleanup
-	DropTable($conn1, $tableName);
-	$stmt1 = null;
-	$conn1 = null;
-
-	EndTest($testName);
+require_once("MsCommon_mid-refactor.inc");
+
+try {
+    $conn1 = connect();
+
+    // Prepare test table
+    $tableName = "pdo_test_table";
+    createTable($conn1, $tableName, array(new ColumnMeta("int", "idx", "NOT NULL PRIMARY KEY"), "txt" => "varchar(20)"));
+    insertRow($conn1, $tableName, array("idx" => 0, "txt" => "String0"));
+    insertRow($conn1, $tableName, array("idx" => 1, "txt" => "String1"));
+    insertRow($conn1, $tableName, array("idx" => 2, "txt" => "String2"));
+    insertRow($conn1, $tableName, array("idx" => 3, "txt" => "String3"));
+
+    // Testing with prepared query
+    logInfo(1, "Testing fetchColumn() ...");
+    $stmt1 = $conn1->query("SELECT COUNT(idx) FROM [$tableName]");
+    var_dump($stmt1->fetchColumn());
+    unset($stmt1);
+
+    logInfo(2, "Testing fetchAll() ...");
+    // ORDER BY doesn't work for encrypted columns
+    // need to fetch all rows first then sort and print
+    if (!isColEncrypted()) {
+        $stmt1 = $conn1->prepare("SELECT idx, txt FROM [$tableName] ORDER BY idx");
+    } else {
+        $stmt1 = $conn1->prepare("SELECT idx, txt FROM [$tableName]");
+    }
+    $stmt1->execute();
+    $data = $stmt1->fetchAll(PDO::FETCH_COLUMN|PDO::FETCH_UNIQUE);
+    if (isColEncrypted()) {
```

Comment:
the fetch style PDO::FETCH_COLUMN returns only one column? With AE, which column does it return here?

Reply:
PDO::FETCH_COLUMN | PDO::FETCH_UNIQUE returns idx as key and txt as value